### PR TITLE
Rework Prometheus metrics scraping

### PR DESCRIPTION
**Pull request by evenh**
This PR adds a new section to the CRD:
```yaml
# Optional settings for how Prometheus compatible metrics should be scraped.
prometheus:
  port: 8181         # *Required* Pod port number or name
  path: "/metrics"   # *Optional* Defaults to /metrics
```

If specified, it will create a matching `ServiceMonitor` and tweak the `NetworkPolicy` to accept scrape traffic from grafana-agent.

### CODE REVIEW

File: controllers/application/service_monitor.go [hunk 1]
```diff
@@ -21,12 +19,16 @@ func (r *ApplicationReconciler) reconcileServiceMonitor(ctx context.Context, app
 		return reconcile.Result{}, nil
 	}
 
-	if !hasMetricsPort(application) {
+	if application.Spec.Prometheus == nil {
 		r.SetControllerFinishedOutcome(ctx, application, controllerName, nil)
 		return reconcile.Result{}, nil
 	}
```
Comment (anderssonw) [inline controllers/application/service_monitor.go:25]: i believe you should add a client.Delete() for the service monitor here. I assume that removing the prometheus config will not remove the resource itself as of now

**Review by anderssonw — CHANGES_REQUESTED** [review]
nice! one small thing regarding deletion of service monitor :)

File: controllers/application/service_monitor.go (hunk 1 above)
Comment (evenh) [reply controllers/application/service_monitor.go]: Good catch!

**Review by anderssonw — APPROVED** [review]
smoothie mcflip

**Review by BardOve — APPROVED** [review]

**Pull request merged**